pkg/agent/openflow/client.go

```diff
@@ -170,61 +177,101 @@ func (c *client) UninstallPodFlows(containerID string) error {
+	} else if err := c.flowOperations.Add(c.l2ForwardCalcFlow(gatewayMAC, gatewayOFPort, cookie.Default)); err != nil {
 		return err
-	} else if err := c.flowOperations.Add(c.localProbeFlow(gatewayAddr)); err != nil {
+	} else if err := c.flowOperations.Add(c.localProbeFlow(gatewayAddr, cookie.Default)); err != nil {
 		return err
 	}
 	return nil
 }
 
 func (c *client) InstallTunnelFlows(tunnelOFPort uint32) error {
-	if err := c.flowOperations.Add(c.tunnelClassifierFlow(tunnelOFPort)); err != nil {
+	if err := c.flowOperations.Add(c.tunnelClassifierFlow(tunnelOFPort, cookie.Default)); err != nil {
 		return err
-	} else if err := c.flowOperations.Add(c.l2ForwardCalcFlow(globalVirtualMAC, tunnelOFPort)); err != nil {
+	}
+	if err := c.flowOperations.Add(c.l2ForwardCalcFlow(globalVirtualMAC, tunnelOFPort, 0)); err != nil {
 		return err
 	}
 	return nil
 }
 
-func (c *client) Initialize() error {
+func (c *client) Initialize(brClient ovsconfig.OVSBridgeClient) error {
 	// Initiate connections to target OFswitch, and create tables on the switch.
 	if err := c.bridge.Connect(maxRetryForOFSwitch, make(chan struct{})); err != nil {
 		return err
 	}
 
+	c.cookieAllocator = cookie.NewAllocator(roundNum(brClient))
 	for _, flow := range c.defaultFlows() {
 		if err := c.flowOperations.Add(flow); err != nil {
 			return fmt.Errorf("failed to install default flows: %v", err)
 		}
 	}
-	if err := c.flowOperations.Add(c.arpNormalFlow()); err != nil {
+	if err := c.flowOperations.Add(c.arpNormalFlow(cookie.Default)); err != nil {
 		return fmt.Errorf("failed to install arp normal flow: %v", err)
 	}
-	if err := c.flowOperations.Add(c.l2ForwardOutputFlow()); err != nil {
+	if err := c.flowOperations.Add(c.l2ForwardOutputFlow(cookie.Default)); err != nil {
 		return fmt.Errorf("failed to install l2 forward output flows: %v", err)
 	}
-	for _, flow := range c.connectionTrackFlows() {
+	for _, flow := range c.connectionTrackFlows(cookie.Default) {
 		if err := c.flowOperations.Add(flow); err != nil {
 			return fmt.Errorf("failed to install connection track flows: %v", err)
 		}
 	}
-	for _, flow := range c.establishedConnectionFlows() {
+	for _, flow := range c.establishedConnectionFlows(cookie.Default) {
 		if err := flow.Add(); err != nil {
 			return fmt.Errorf("failed to install flows to skip established connections: %v", err)
 		}
 	}
 	return nil
 }
+
+func roundNum(bridgeClient ovsconfig.OVSBridgeClient) uint64 {
+	var num uint64
+	rand.Seed(time.Now().UnixNano())
+	// read the round number
+	extIDs, ovsErr := bridgeClient.GetExternalIDs()
+	if ovsErr != nil {
+		klog.Warningf("Getting last round number failed: %s", ovsErr.Error())
+		num = rand.Uint64()
+	} else if roundNumValue, ok := extIDs[roundNumKey]; ok {
+		var err error
+		num, err = strconv.ParseUint(roundNumValue, 10, 64)
+		if err != nil {
+			klog.Warningf("Parsing last round number failed: %v", err)
+			num = rand.Uint64()
+		}
+		if num == (1<<cookie.BitwidthRound)-1 {
```

Comment:
I think this is redundant with the modulo operation below: `num = rand.Uint64() % (1 << cookie.BitwidthRound)`. I suggest you replace this `if` block with `num += 1`, and let the modulo operation below take care of the wrap-around.

Reply:
Thanks, I would change as you said